### PR TITLE
Detect dark mode and load the proper cds theme

### PR DESCRIPTION
**Pull request by antgamdia**
_Depends on #2443_ 

### Description of the change

This PR changes simply supports dark mode **in cds** (not clr). As a result, very few cds-styled components will get the dark theme when the user's browser is configured in dark mode.

### Benefits

This PR paves the way for the upgrading process from clarity v4 to v5 and dark mode.

### Possible drawbacks

This PR (and some of the subsequent ones) are aimed at allowing dark mode in clarity v5. Until every PR becomes merged, some css inconsistencies might occur.

### Applicable issues

  - related #2372
  - related #2371 

### Additional information

The whole set of PRs includes:

- Remove hardcoded css, delegate colors in the cds variables (which is more customizable through themes), minor css fixes.
- Bump up dependencies. Minor fixes.
  - Update modals
  - Update layout
  - Add media query to detect dark mode (for cds)
  - Add dynamic dark mode for clr

### CODE REVIEW

**Review by antgamdia — COMMENTED** [review]
It's a temporary change (superseded by #2444), so we can merge it or even close, as you may prefer

**Review by andresmgot — APPROVED** [review]
Can we add a test for this?

File: dashboard/src/containers/Root.tsx [hunk 1]
```diff
@@ -18,9 +18,16 @@ async function initLocale() {
 
 function Root() {
   const [i18nConfig, setI18nConfig] = useState(I18n.getDefaultConfig());
+  // Currently we just support 'dark' and 'light' themes
+  // TODO(agamez): add manual override for this setting
+  const theme =
+    window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches
+      ? "dark"
+      : "light";
   useEffect(() => {
```
Comment (andresmgot) [inline dashboard/src/containers/Root.tsx:27]: also, better to use a different `useEffect` to set this. I believe there is no relation between `initLocale` and setting the cds-theme property.

**Comment by antgamdia** [comment]
These changes are being superseded by https://github.com/kubeapps/kubeapps/pull/2450 (taking into account this PR's comments), so let's close it.

**Pull request closed without merging**